metrics/ci_integrations/lib/client/github_actions/github_actions_client.dart

```diff
@@ -0,0 +1,256 @@
+  ///
+  /// [page] can be used to set the number of a page to retrieve.
+  /// [page] defaults to the `1`st page.
+  Future<InteractionResult<List<WorkflowRun>>> fetchWorkflowRuns(
+    String workflowFileName, {
+    RunStatus status = RunStatus.completed,
+    int perPage = 100,
+    int page = 1,
+  }) {
+    const statusMapper = RunStatusMapper();
+
+    final queryParameters = {
+      'status': statusMapper.unmap(status),
+      'per_page': '$perPage',
+      'page': '$page',
+    };
+
+    final url = UrlUtils.buildUrl(
+      baseUrl,
+      path: 'workflows/$workflowFileName/runs',
+      queryParameters: queryParameters,
+    );
+
+    return _handleResponse<List<WorkflowRun>>(
+      _client.get(url, headers: headers),
+      (Map<String, dynamic> json) {
+        final runList =
+            json == null ? null : json['workflow_runs'] as List<dynamic>;
+
+        return InteractionResult.success(
+          result: WorkflowRun.listFromJson(runList),
+        );
+      },
+    );
+  }
+
+  /// Retrieves a workflow run duration by the given [runId].
+  ///
+  /// Results with a [WorkflowRunDuration].
+  Future<InteractionResult<WorkflowRunDuration>> fetchRunDuration(int runId) {
+    final url = UrlUtils.buildUrl(
+      baseUrl,
+      path: 'runs/$runId/timing',
+    );
+
+    return _handleResponse<WorkflowRunDuration>(
+      _client.get(url, headers: headers),
+      (Map<String, dynamic> json) {
+        return InteractionResult.success(
+          result: WorkflowRunDuration.fromJson(json),
+        );
+      },
+    );
+  }
+
+  /// Retrieves a list of workflow run artifacts by the given [runId].
+  ///
+  /// Results with a [List] of [WorkflowRunArtifact]s.
+  ///
+  /// [perPage] can be used to set the number of artifacts per page.
+  /// [perPage] defaults to `100` artifacts per page.
+  /// The maximum value of [perPage] is 100.
+  ///
+  /// [page] can be used to set the number of a page to retrieve.
+  /// [page] defaults to the `1`st page.
+  Future<InteractionResult<List<WorkflowRunArtifact>>> fetchRunArtifacts(
+    int runId, {
+    int perPage = 100,
+    int page = 1,
+  }) {
+    final queryParameters = {
+      'per_page': '$perPage',
+      'page': '$page',
+    };
+
+    final url = UrlUtils.buildUrl(
+      baseUrl,
+      path: 'runs/$runId/artifacts',
+      queryParameters: queryParameters,
+    );
+
+    return _handleResponse<List<WorkflowRunArtifact>>(
+      _client.get(url, headers: headers),
+      (Map<String, dynamic> json) {
+        final artifactList =
+            json == null ? null : json['artifacts'] as List<dynamic>;
+
+        return InteractionResult.success(
+          result: WorkflowRunArtifact.listFromJson(artifactList),
+        );
+      },
+    );
+  }
+
+  /// Retrieves a workflow run artifact by the given download [url].
+  ///
+  /// Results with a [Uint8List] that contains the artifact's bytes.
+  Future<InteractionResult<Uint8List>> downloadRunArtifact(String url) async {
```

Comment:
```suggestion
  Future<InteractionResult<Uint8List>> downloadRunArtifactZip(String url) async {
```
Would it be more clear for returning type?